c2corg_api/tests/views/test_document_delete.py

```diff
@@ -643,3 +644,118 @@ def image_service_mock(url, request):
+
+    def test_delete_locale_non_existing_document(self):
+        self._delete_locale(-9999999, 'fr', 400)
+
+    def test_delete_locale_non_existing_locale(self):
+        self._delete_locale(self.waypoint1.document_id, 'de', 400)
+
+    def test_delete_locale_main_waypoint_with_only_one_locale(self):
+        response = self._delete_locale(self.waypoint1.document_id, 'fr', 400)
+        self.assertErrorsContain(
+            response.json, 'document_id',
+            'This waypoint cannot be deleted because it is a main waypoint.')
+
+    def _test_delete_locale(self, document_id, lang):
+        available_langs = get_available_langs(document_id)
+
+        initial_other_locale_archives_count = self.session. \
+            query(ArchiveDocumentLocale). \
+            filter(ArchiveDocumentLocale.document_id == document_id). \
+            filter(ArchiveDocumentLocale.lang != lang).count()
+
+        initial_other_locale_versions_count = self.session. \
+            query(DocumentVersion). \
+            filter(DocumentVersion.document_id == document_id). \
+            filter(DocumentVersion.lang != lang).count()
+
+        self._delete_locale(document_id, lang, 200)
+
+        # Check that deleted locale is no longer available
+        available_langs.remove(lang)
+        final_available_langs = get_available_langs(document_id)
+        # if the document had only one locale, it is deleted and
+        # get_available_langs() returns None
+        self.assertEqual(not final_available_langs, not available_langs)
+        if available_langs:
+            self.assertCountEqual(final_available_langs, available_langs)
+
+        # Check that archives have been removed too
+        count = self.session.query(ArchiveDocumentLocale). \
+            filter(ArchiveDocumentLocale.document_id == document_id). \
+            filter(ArchiveDocumentLocale.lang == lang).count()
+        self.assertEqual(0, count)
+
+        # but not other locale archives
+        other_locale_archives_count = self.session. \
+            query(ArchiveDocumentLocale). \
+            filter(ArchiveDocumentLocale.document_id == document_id). \
+            filter(ArchiveDocumentLocale.lang != lang).count()
+        self.assertEqual(other_locale_archives_count,
+                         initial_other_locale_archives_count)
+
+        # Check that the document versions table is cleared
+        count = self.session.query(DocumentVersion). \
+            filter(DocumentVersion.document_id == document_id). \
+            filter(DocumentVersion.lang == lang).count()
+        self.assertEqual(0, count)
+
+        # but not for other locales
+        other_locale_versions_count = self.session. \
+            query(DocumentVersion). \
+            filter(DocumentVersion.document_id == document_id). \
+            filter(DocumentVersion.lang != lang).count()
+        self.assertEqual(other_locale_versions_count,
+                         initial_other_locale_versions_count)
+
```

Comment:
This is not strictly needed, but match with the tests policy we have used on the API, so I've noted that is not done. If you think it's not necessary, feel free to merge.